test/e2e/service.go

```diff
@@ -1991,3 +2068,46 @@ func (t *ServiceTestFixture) Cleanup() []error {
 
 	return errs
 }
+
+// launchEchoserverPod launches a pod serving http on port 8080 to act
+// as the target for source IP preservation checks. The client's source ip would
+// be echoed back by the web server.
+func (j *ServiceTestJig) launchEchoserverPod(f *framework.Framework, nodeName, podName string) {
+	containerName := fmt.Sprintf("%s-container", podName)
+	port := 8080
+	pod := &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			Name:   podName,
+			Labels: j.Labels,
+		},
+		Spec: api.PodSpec{
+			Containers: []api.Container{
+				{
+					Name:  containerName,
+					Image: "gcr.io/google_containers/echoserver:1.4",
+					Ports: []api.ContainerPort{{ContainerPort: int32(port)}},
+				},
+			},
+			NodeName:      nodeName,
+			RestartPolicy: api.RestartPolicyNever,
+		},
+	}
+	By(fmt.Sprintf("Creating echo server pod %q in namespace %q", pod.Name, f.Namespace.Name))
+	podClient := f.Client.Pods(f.Namespace.Name)
+	_, err := podClient.Create(pod)
+	framework.ExpectNoError(err)
+	framework.ExpectNoError(f.WaitForPodRunning(podName))
+	By(fmt.Sprintf("Echo server pod %q in namespace %q running", pod.Name, f.Namespace.Name))
+}
+
+// LaunchHostExecPod launches a hostexec pod in the given namespace and node
+// waits until it's Running
+func launchHostExecPodOnNode(client *client.Client, ns, nodeName, podName string) *api.Pod {
+	hostExecPod := framework.NewHostExecPodSpec(ns, podName)
```

Comment:
I think you should not use HostExecPod. You should use a normal pod without hostnetwork.